src/jdk.jlink/share/classes/jdk/tools/jlink/builder/ImageBuilder.java

```diff
@@ -80,8 +80,9 @@ public default void storeFiles(ResourcePool content) {
      * Gets the platform of the image.
      *
      * @return {@code Platform} object representing the platform of the image
+     * @throws UnsupportedOperationException if this method is not implemented by the ImageBuilder
      */
     public default Platform getTargetPlatform() {
-        return Platform.UNKNOWN;
+        throw new UnsupportedOperationException("Builder does not define getTargetPlatform");
```

Comment:
I think it is OK to change this implementation to now throw this exception, since this is an internal interface (from what I can see in the module definition).